Source/items.h

```diff
@@ -240,7 +240,7 @@ struct Item {
 	bool _iStatFlag = false;
 	_item_indexes IDidx = IDI_NONE;
 	uint32_t dwBuff = 0;
-	uint32_t _iDamAcFlags = 0;
+	ItemSpecialEffectHf _iDamAcFlags = ItemSpecialEffectHf::None;

```

Comment:
If this is a *set* of flags then this is not the correct type for it.
`std::underlying_type<ItemSpecialEffectHf>::type` is more appropriate.

Typing a set as the enum type breaks static analysis of switch statements.

Reply:
Can you clarify what I'm supposed to do here? Not familiar with this one, and it looks like it doesn't like it when I rename do the variable declarations accordingly: 
![image](https://user-images.githubusercontent.com/6080734/156923699-7c42782e-9767-47b8-93a5-41e631d3e1fd.png)

Reply:
I was under the impression that using the enum as the type was suitable for flags enums given the expected values fall within the expected range of the enum (or the underlying type is fixed, which bypasses the issue entirely). Using a flag enum in a switch statement seems wrong anyway, it'd only test for an exact value instead of whether a given flag is set?

Using the underlying type directly makes it more difficult to assign/set/remove flags but that could be worked around without much drama if we wanted to go that way.

Reply:
I suppose it's OK to use the enum class directly if it's verbose otherwise. This does mean it is not possible to distinguish a single value from a set based on the type alone, but I guess `flags` is there in the name so it's not a major issue in practice.

Reply:
Sounds good. I'll leave as is.